msprime/mutations.py

```diff
@@ -63,6 +69,113 @@ def asdict(self):
             self.__init__).parameters.keys() if hasattr(self, key)}
 
 
+class FiniteSites(object):
+    """
+    """
+    def __init__(self, seed, rate_map, alphabet, mutation_matrix, alphabet_distribution):
+        if alphabet not in [BINARY, NUCLEOTIDES]:
+            raise ValueError("Bad alphabet")
+        seed = int(seed)
+        rng = _msprime.RandomGenerator(seed)
+        alphabet_length = {BINARY: 2, NUCLEOTIDES: 4}[alphabet]
+        try:
```

Comment:
How about we coerce this into a numpy array and check that?
```
mutatin_matrix = np.array(mutation_matrix)
if len(mutation_matrix.shape)! = 2:
     raise ValueError("Mutaiton matrix must be a 2D array") 
# etc
```
We'll be casting it to a numpy array when we're converting to C data structures anyway, so might as well use the convenience of numpy here.